components/cluster/command/patch.go

```diff
@@ -186,13 +187,23 @@ func overwritePatch(clusterName, comp, packagePath string) error {
 	if err := os.MkdirAll(spec.ClusterPath(clusterName, spec.PatchDirName), 0755); err != nil {
 		return err
 	}
+
 	checksum, err := tiuputils.Checksum(packagePath)
 	if err != nil {
 		return err
 	}
+
 	tg := spec.ClusterPath(clusterName, spec.PatchDirName, comp+"-"+checksum[:7]+".tar.gz")
+	if utils.IsExist(tg) {
```

Comment:
I think we can skip the `CopyFile` is the target file exists.

Reply:
Not a good idea, for example:
```sh
# In tidb code directory
make tidb
tar -czf tidb.tar.gz tidb-server
tiup cluster patch test-cluster $(pwd)/tidb.tar.gz -R tidb

# Test tidb in the cluster found a bug, changed the code
make tidb
tar -czf tidb.tar.gz tidb-server
tiup cluster patch test-cluster $(pwd)/tidb.tar.gz -R tidb
```

This will skip the second patch.

Reply:
The checksum will be different.